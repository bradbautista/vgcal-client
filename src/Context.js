import React from 'react'

const Context = React.createContext({
    releases: [],    
})

export default Context