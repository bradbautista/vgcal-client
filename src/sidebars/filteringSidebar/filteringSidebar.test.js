import React from 'react';
import ReactDOM from 'react-dom';
import FiltersList from './filtersList';
import FavoritesList from './favoritesList';
import Context from '../../Context';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

it('Renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Context.Provider value={ { favorites: [], checkboxes: [] }}><FiltersList /><FavoritesList/></Context.Provider>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Renders the filtering sidebar list as expected when there are no favorites', () => {
    
    const tree = renderer
        .create(
            
            <BrowserRouter>
                <Context.Provider value={ { favorites: [], checkboxes: [] }}>
                    <FiltersList />
                    <FavoritesList/>
                </Context.Provider>
            </BrowserRouter>
            )
        .toJSON();
    expect(tree).toMatchSnapshot();
        
})

it('Renders the filtering sidebar list as expected when there are favorites', () => {

    const favorites = [{"platforms":"Nintendo Switch","image":"https://giantbomb1.cbsistatic.com/uploads/original/0/30/3177321-job.png","gameUrl":"/game/Good-Job!","gameTitle":"Good Job!","releaseDate":"March 26, 2020","description":"Solve workplace problems with the power of physics in this action puzzle game for the Nintendo Switch."}]
    
    const tree = renderer
        .create(
            
            <BrowserRouter>
                <Context.Provider value={ { favorites: favorites, checkboxes: [] }}>
                    <FiltersList />
                    <FavoritesList/>
                </Context.Provider>
            </BrowserRouter>
            )
        .toJSON();
    expect(tree).toMatchSnapshot();
        
})