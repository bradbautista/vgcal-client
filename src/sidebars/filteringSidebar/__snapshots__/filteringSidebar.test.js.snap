// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the filtering sidebar list as expected when there are favorites 1`] = `
Array [
  <div
    className="filtering-container"
  >
    <h2
      className="filters-header"
    >
      Filtering options
    </h2>
    <div>
      Note: If you check PC and Switch, you will get games which are coming out on PC 
      <em>
        and
      </em>
       Switch, not PC 
      <em>
        or
      </em>
       Switch.
    </div>
    <label
      className="checkbox-wrapper"
      htmlFor="Xbox"
    >
      <input
        checked={false}
        id="Xbox"
        name="Xbox"
        onChange={[Function]}
        type="checkbox"
        value="Xbox"
      />
      <span
        className="checkmark"
      />
      Xbox
    </label>
    <label
      className="checkbox-wrapper"
      htmlFor="PlayStation"
    >
      <input
        checked={false}
        id="PlayStation"
        name="PlayStation"
        onChange={[Function]}
        type="checkbox"
        value="PlayStation"
      />
      <span
        className="checkmark"
      />
      PlayStation
    </label>
    <label
      className="checkbox-wrapper"
      htmlFor="PC"
    >
      <input
        checked={false}
        id="PC"
        name="PC"
        onChange={[Function]}
        type="checkbox"
        value="PC"
      />
      <span
        className="checkmark"
      />
      PC
    </label>
    <label
      className="checkbox-wrapper"
      htmlFor="Switch"
    >
      <input
        checked={false}
        id="Switch"
        name="Switch"
        onChange={[Function]}
        type="checkbox"
        value="Switch"
      />
      <span
        className="checkmark"
      />
      Switch
    </label>
    <label
      className="checkbox-wrapper"
      htmlFor="Mac"
    >
      <input
        checked={false}
        id="Mac"
        name="Mac"
        onChange={[Function]}
        type="checkbox"
        value="Mac"
      />
      <span
        className="checkmark"
      />
      Mac
    </label>
    <label
      className="checkbox-wrapper"
      htmlFor="Android"
    >
      <input
        checked={false}
        id="Android"
        name="Android"
        onChange={[Function]}
        type="checkbox"
        value="Android"
      />
      <span
        className="checkmark"
      />
      Android
    </label>
    <label
      className="checkbox-wrapper"
      htmlFor="iPhone"
    >
      <input
        checked={false}
        id="iPhone"
        name="iPhone"
        onChange={[Function]}
        type="checkbox"
        value="iPhone"
      />
      <span
        className="checkmark"
      />
      iPhone
    </label>
    <label
      className="checkbox-wrapper"
      htmlFor="iPad"
    >
      <input
        checked={false}
        id="iPad"
        name="iPad"
        onChange={[Function]}
        type="checkbox"
        value="iPad"
      />
      <span
        className="checkmark"
      />
      iPad
    </label>
  </div>,
  <div
    className="favorites-container"
  >
    <h2
      className="favorites-header"
    >
      Your favorites
    </h2>
    <p>
      To delete a game from the list, click on it.
    </p>
    <ul
      className="favorites-list"
    >
      <li
        className="favorite"
        onClick={[Function]}
      >
        <img
          alt="Good Job!"
          className="favorite-image"
          src="https://giantbomb1.cbsistatic.com/uploads/original/0/30/3177321-job.png"
        />
        <span
          className="favorite-title"
        >
          Good Job!
        </span>
      </li>
    </ul>
    <button
      className="export-button unfavorited"
      disabled={false}
      onClick={[Function]}
    >
      Export to ICS
    </button>
  </div>,
]
`;

exports[`Renders the filtering sidebar list as expected when there are no favorites 1`] = `
Array [
  <div
    className="filtering-container"
  >
    <h2
      className="filters-header"
    >
      Filtering options
    </h2>
    <div>
      Note: If you check PC and Switch, you will get games which are coming out on PC 
      <em>
        and
      </em>
       Switch, not PC 
      <em>
        or
      </em>
       Switch.
    </div>
    <label
      className="checkbox-wrapper"
      htmlFor="Xbox"
    >
      <input
        checked={false}
        id="Xbox"
        name="Xbox"
        onChange={[Function]}
        type="checkbox"
        value="Xbox"
      />
      <span
        className="checkmark"
      />
      Xbox
    </label>
    <label
      className="checkbox-wrapper"
      htmlFor="PlayStation"
    >
      <input
        checked={false}
        id="PlayStation"
        name="PlayStation"
        onChange={[Function]}
        type="checkbox"
        value="PlayStation"
      />
      <span
        className="checkmark"
      />
      PlayStation
    </label>
    <label
      className="checkbox-wrapper"
      htmlFor="PC"
    >
      <input
        checked={false}
        id="PC"
        name="PC"
        onChange={[Function]}
        type="checkbox"
        value="PC"
      />
      <span
        className="checkmark"
      />
      PC
    </label>
    <label
      className="checkbox-wrapper"
      htmlFor="Switch"
    >
      <input
        checked={false}
        id="Switch"
        name="Switch"
        onChange={[Function]}
        type="checkbox"
        value="Switch"
      />
      <span
        className="checkmark"
      />
      Switch
    </label>
    <label
      className="checkbox-wrapper"
      htmlFor="Mac"
    >
      <input
        checked={false}
        id="Mac"
        name="Mac"
        onChange={[Function]}
        type="checkbox"
        value="Mac"
      />
      <span
        className="checkmark"
      />
      Mac
    </label>
    <label
      className="checkbox-wrapper"
      htmlFor="Android"
    >
      <input
        checked={false}
        id="Android"
        name="Android"
        onChange={[Function]}
        type="checkbox"
        value="Android"
      />
      <span
        className="checkmark"
      />
      Android
    </label>
    <label
      className="checkbox-wrapper"
      htmlFor="iPhone"
    >
      <input
        checked={false}
        id="iPhone"
        name="iPhone"
        onChange={[Function]}
        type="checkbox"
        value="iPhone"
      />
      <span
        className="checkmark"
      />
      iPhone
    </label>
    <label
      className="checkbox-wrapper"
      htmlFor="iPad"
    >
      <input
        checked={false}
        id="iPad"
        name="iPad"
        onChange={[Function]}
        type="checkbox"
        value="iPad"
      />
      <span
        className="checkmark"
      />
      iPad
    </label>
  </div>,
  <div
    className="favorites-container"
  >
    <h2
      className="favorites-header"
    >
      Your favorites
    </h2>
    
    <ul
      className="favorites-list"
    >
      <li
        className="favorites-prompt"
      >
        Add games to your favorites list and you can export an ics file to add the release dates to your personal calendar.
      </li>
    </ul>
    <button
      className="export-button"
      disabled={true}
      onClick={[Function]}
    >
      Export to ICS
    </button>
  </div>,
]
`;
